addon/components/select-2.js

```diff
@@ -90,7 +92,11 @@ var Select2Component = Ember.Component.extend({
 
     // override select2's default id fetching behavior
     options.id = (function(e) {
-      return (e === undefined) ? null : get(e, optionIdPath);
+      var id = (e === undefined) ? null : get(e, optionIdPath);
+      if (self.get('tags')) {
+        return id.toLowerCase();
```

Comment:
I am forcing the lowercase of the ID here, since you don't want tags that only differ in uppercase or lowercase and are still added.